dali/test/python/test_data_containers.py

```diff
@@ -77,31 +77,34 @@ def __init__(self, batch_size, num_threads, device_id, num_gpus, data_paths):
                                         })
 
     def define_graph(self):
-        inputs = self.input()
+        inputs = self.input(name="Reader")
         images = inputs["image/encoded"]
         labels = inputs["image/class/label"]
         return self.base_define_graph(images, labels)
 
 test_data = {
-            FileReadPipeline: [["/data/imagenet/", "/data/imagenet/train-jpeg_map.txt", 1281167],
-                               ["/data/imagenet/", "/data/imagenet/val-jpeg_map.txt", 50000]],
-            MXNetReaderPipeline: [["/data/imagenet/train-480-val-256-recordio/train.rec", "/data/imagenet/train-480-val-256-recordio/train.idx", 1281167],
-                                   ["/data/imagenet/train-480-val-256-recordio/val.rec", "/data/imagenet/train-480-val-256-recordio/val.idx", 50000]],
-            CaffeReadPipeline: [["/data/imagenet/train-lmdb-256x256", 1281167],
-                                 ["/data/imagenet/val-lmdb-256x256", 50000]],
-            Caffe2ReadPipeline: [["/data/imagenet/train-c2lmdb-480", 1281167],
-                                  ["/data/imagenet/val-c2lmdb-256", 50000]],
-            TFRecordPipeline: [["/data/imagenet/train-val-tfrecord-480/train-*", "/data/imagenet/train-val-tfrecord-480.idx/train-*", 1281167]]
+            FileReadPipeline: [["/data/imagenet/train-jpeg"],
+                               ["/data/imagenet/val-jpeg"]]
+            MXNetReaderPipeline: [["/data/imagenet/train-480-val-256-recordio/train.rec", "/data/imagenet/train-480-val-256-recordio/train.idx"],
+                                   ["/data/imagenet/train-480-val-256-recordio/val.rec", "/data/imagenet/train-480-val-256-recordio/val.idx"]],
+            CaffeReadPipeline: [["/data/imagenet/train-lmdb-256x256"],
+                                 ["/data/imagenet/val-lmdb-256x256"]],
+            Caffe2ReadPipeline: [["/data/imagenet/train-c2lmdb-480"],
+                                  ["/data/imagenet/val-c2lmdb-256"]],
+            TFRecordPipeline: [["/data/imagenet/train-val-tfrecord-480/train-*", "/data/imagenet/train-val-tfrecord-480.idx/train-*"]]
             }
 
-N = 4               # number of GPUs
+N = 1               # number of GPUs
 BATCH_SIZE = 2048   # batch size
 LOG_INTERVAL = 200 // BATCH_SIZE + 1
 
 for pipe_name in test_data.keys():
     data_set_len = len(test_data[pipe_name])
     for i, data_set in enumerate(test_data[pipe_name]):
-        iters = data_set[-1]
+        pipes = [pipe_name(batch_size=BATCH_SIZE, num_threads=4, device_id = n, num_gpus = N, data_paths = data_set) for n in range(N)]
+        [pipe.build() for pipe in pipes]
+
+        iters = pipes[0].epoch_size("Reader")
```

Comment:
Can we here check whether all of the iterators report the same size?

Reply:
Otherwise lgtm

Reply:
Done